stripe/src/main/java/com/stripe/android/paymentsheet/PaymentOptionsViewModel.kt

```diff
@@ -21,6 +21,35 @@ internal class PaymentOptionsViewModel(
     config = args.config,
     prefsRepository = prefsRepository
 ) {
+    // This is used only for the case where we need to jump straight to the add card when
+    // the last card added was unsaved.  We have this variable so that when you press
+    // the back button it doesn't launch the add card again.
+    private var _trampolinedToCard = true
+
+    fun resolveFinalFragmentOnPaymentSelection(config: FragmentConfig) {
+        if (isLastSelectionNewUnsavedCard(newCard)) {
+
+            if (_trampolinedToCard) {
+                // If we press the back button we don't want to open the add card again
+                _trampolinedToCard = false
+                transitionTo(
+                    TransitionTarget.AddPaymentMethodFull(config)
+                )
```

Comment:
Should this logic only apply when using a customer?

Reply:
In iPhone:
1. Use the PaymentSheet Test Playground
2. Use customer = guestmode
3. Enter a new payment method and Add
4. Click on payment methods again, it will show the card details.
So it applies if you are using a customer or not.